velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +367,64 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
+            RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    // apply() will not be invoked if the pattern was invalid or the selection
+    // is empty.
+    checkForBadPattern(re_);
+
+    FlatVector<bool>& result =
+        ensureWritableBool(rows, context->pool(), resultRef);
+
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto toSearch = decodedArgs.at(0);
+    if (toSearch->isIdentityMapping()) {
+      auto rawStrings = toSearch->data<StringView>();
+      rows.applyToSelected([&](vector_size_t i) {
+        if (!validPattern_) {
```

Comment:
Let's move this checkout outside of the loop and run it at the very beginning of the apply function before doing any work.

Also, we don't need a macro here. This is a user-visible error and file and line number information would not be useful.

```
          auto error = std::make_exception_ptr(
              std::invalid_argument("Escape character must be followed..."));
          rows.applyToSelected(
              [&](auto row) { context->setError(row, error); });
```